### PR TITLE
doc,man: Typos found by codespell

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Mostly in documentation and user visible text, but also code comments.

Left out `src/` for now.

Will not modify most of `releases` and `changelog`, keeping reference to past commits with their typos.

**Edit:** reworked this PR to keep only changes under `doc/` and `man/`

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [X] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
I'd suggest to make a dedicated PR for just for all changes in doc/ , as fixing typos there actually has a real value. I'd leave this PR here for the remaining changes outside of src/

**Comment by wjwithagen** [comment]
There is that discussion of a time back where it was decided that fixing typos in comments and code was not worth a PR in itself. 
If things need fixing, do it while actually changing the code at that spot.
This to prevent pollution in github attribution.

**Comment by sebastian-philipp** [comment]
> 

Right. Still, there are a few changes that affect example code within comments and fixing those might actually make sense.

**Comment by wjwithagen** [comment]
Sure, but the conclussion was that typos without significant purpose shoudl not be done in a fly-over.

**Comment by DimitriPapadopoulos** [comment]
Indeed I have left out `src/` and mainly fixed examples, documentation, and user-visible text.

I may have fixed some typos in code comments, but that's because they generate noise in spell checking: [codespell](https://github.com/codespell-project/codespell) does not make any difference between code, user-visible-text in code, code comments and documentation.

**Comment by wjwithagen** [comment]
I would still restrict this commit to doc-only.
Don't do cmake, example, *.yml or anything that looks like code.

**Comment by DimitriPapadopoulos** [comment]
Sure, but can I a try a different PR for the examples outside `doc/`, or will such a PR be certainly rejected?

**Comment by wjwithagen** [comment]
This lies much more subtle.
I think that typos that will be shown to users are fine to submit.

Others I do not expect get accepted with also some actual code changes, and then only when code and type are related.
Note that there is more to developer code than just 'src/'
What I saw from your patchset, I would (also) not submit most changes to things like ` *.yml *.py *.cmake`
But then again a type in a print in `*.py` would be oke, but in a comment it serves little purpose.

I would also suggest to not make this a PR of this size. It shows also from the fact that you have not decided on which component you are fixing. Doc would make sense, but theen only for things in `doc/` of `man/` 
This also shows from the fact that your PR got a lot of labels auto-assigned. Probably got to break it into several pieces.

**Comment by DimitriPapadopoulos** [comment]
I have removed all changes but those under `doc/` and `man/`.

File: doc/dev/radosgw/s3_compliance.rst [hunk 1]
```diff
@@ -290,7 +290,7 @@ S3 Documentation reference : http://docs.aws.amazon.com/AmazonS3/latest/API/REST
 +---------+---------------------------+------------+---------------------------------------------------------------------------------------------------------+-------------+
 | PUT     | Object copy               | Yes        |                                                                                                         |             |
 +---------+---------------------------+------------+---------------------------------------------------------------------------------------------------------+-------------+
-| PUT     | Initate multipart upload  | Yes        |                                                                                                         |             |
+| PUT     | Initiate multipart upload | Yes        |                                                                                                         |             |
```
Comment (sebastian-philipp) [inline doc/dev/radosgw/s3_compliance.rst:293]: can you verify the table still renders properly?

File: doc/releases/emperor.rst [hunk 2]
```diff
@@ -194,7 +194,7 @@ Upgrading from v0.67 Dumpling
   2012) and store radosgw data, the auto-upgrade from TMAP to OMAP
   objects has been disabled.  Before upgrading, make sure that any
   buckets created on pre-argonaut releases have been modified (e.g.,
-  by PUTing and then DELETEing an object from each bucket).  Any
+  by PUTting and then DELETEing an object from each bucket).  Any
```
Comment (sebastian-philipp) [inline doc/releases/emperor.rst:197]: I'd keep the Emperor release notes untouched.

File: doc/releases/emperor.rst [hunk 3]
```diff
@@ -420,7 +420,7 @@ Upgrading
...
```
Comment (sebastian-philipp) [inline doc/releases/emperor.rst:423]: I'd keep the Emperor release notes untouched.

**Review by sebastian-philipp — COMMENTED** [review]
lgtm otherwise

File: doc/releases/emperor.rst (hunk 2 above)
Comment (DimitriPapadopoulos) [reply doc/releases/emperor.rst:197]: Fixed.

File: doc/releases/emperor.rst (hunk 3 above)
Comment (DimitriPapadopoulos) [reply doc/releases/emperor.rst:423]: Fixed.

**Comment by sebastian-philipp** [comment]
would you be ok with changing the title of your commit to `doc,man: Fix typos found by codespell` or something similar?

File: doc/dev/radosgw/s3_compliance.rst (hunk 1 above)
Comment (DimitriPapadopoulos) [reply doc/dev/radosgw/s3_compliance.rst:293]: I was expecting to find the documentation as a CI job artefact, but I cannot find anything in the [Docs: build check](https://jenkins.ceph.com/job/ceph-pr-docs/57704/) CI job.

Comment (DimitriPapadopoulos) [reply doc/dev/radosgw/s3_compliance.rst:293]: Got it it's the [docs/readthedocs.org:ceph](https://ceph--44249.org.readthedocs.build/en/44249/) job instead.

Points to the documentation here:
https://ceph--44249.org.readthedocs.build/en/44249/

Comment (DimitriPapadopoulos) [reply doc/dev/radosgw/s3_compliance.rst:293]: The table renders as expected, as far as I can see:
https://ceph--44249.org.readthedocs.build/en/44249/dev/radosgw/s3_compliance/#operations-on-objects

**Review by sebastian-philipp — APPROVED** [review]

File: doc/start/documenting-ceph.rst [hunk 4]
```diff
@@ -667,7 +667,7 @@ improves the readability of the document in a command line interface.
 
 
 .. _Python Sphinx: http://sphinx-doc.org
-.. _resturcturedText: http://docutils.sourceforge.net/rst.html
+.. _restructuredText: http://docutils.sourceforge.net/rst.html
```
Comment (sebastian-philipp) [inline doc/start/documenting-ceph.rst:670]: this probably broke the link. Note this is not a user-visible string, but an identifier.

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: doc/start/documenting-ceph.rst (hunk 4 above)
Comment (DimitriPapadopoulos) [reply doc/start/documenting-ceph.rst]: Ah indeed, there's no point fixing this so I've reverted this specific change.

That said, a `grep` search for `resturcturedText` doesn't show any other occurrence in this files or elsewhere. Also, none of the occurrences of `reStructuredText`in this page is a link:
https://docs.ceph.com/en/latest/start/documenting-ceph/

**Review by callithea — APPROVED** [review]

**Comment by callithea** [comment]
jenkins test dashboard

File: doc/start/os-recommendations.rst [hunk 5]
```diff
@@ -14,12 +14,12 @@ Linux Kernel
...
 
   If you are using the kernel client to map RBD block devices or mount
-  CephFS, the general advice is to use a "stable" or "longterm
+  CephFS, the general advice is to use a "stable" or "long term
   maintenance" kernel series provided by either http://kernel.org or
   your Linux distribution on any client hosts.
 
   For RBD, if you choose to *track* long-term kernels, we currently recommend
-  4.x-based "longterm maintenance" kernel series or later:
+  4.x-based "long term maintenance" kernel series or later:
```
Comment (sebastian-philipp) [inline doc/start/os-recommendations.rst:22]: https://www.kernel.org/category/releases.html

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]

File: doc/start/os-recommendations.rst (hunk 5 above)
Comment (DimitriPapadopoulos) [reply doc/start/os-recommendations.rst:22]: You would rather use the term coined by the Linux Kernel Organization then, even if it's incorrect - I cannot find "longterm" in any dictionary. It's OK, I'll revert that. Besides, it appears "long-term" would be the correct word here, not "long term":
* [Which is correct: long-term or long term?](https://whatis.techtarget.com/feature/Which-is-correct-long-term-or-long-term)
* [Is it long term or long-term?](https://www.future-perfect.co.uk/grammar-tips/easy-mistakes/is-it-long-term-or-long-term)
* Merriam-Webster entry for [long-term](https://www.merriam-webster.com/dictionary/long-term)

File: doc/releases/luminous.rst [hunk 6]
```diff
@@ -8,7 +8,7 @@ luminous squid (watasenia scintillans, aka firefly squid).
 v12.2.13 Luminous
 =================
 
-This is the 13th bug fix release of the Luminous v12.2.x long term stable
+This is the 13th bug fix release of the Luminous v12.2.x long-term stable
```
Comment (sebastian-philipp) [inline doc/releases/luminous.rst:11]: can you revert those? I don't think you're improving things here.

**Review by sebastian-philipp — CHANGES_REQUESTED** [review]
haha, every time I'm reviewing this, you're coming up with new things I don't agree with. Let's revert this "long term" "long-term" change without introducing new changes please. This is getting tedious.

File: doc/releases/luminous.rst (hunk 6 above)
Comment (DimitriPapadopoulos) [reply doc/releases/luminous.rst]: Sure, done.

**Review by sebastian-philipp — APPROVED** [review]

**Comment by sebastian-philipp** [comment]
Thank you for your contribution!

**Pull request merged**